homeassistant/components/ihc/ihcdevice.py

```diff
@@ -41,21 +65,52 @@ def name(self):
         """Return the device name."""
         return self._name
 
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return f"ihc{self.controller_id}{self.ihc_id}"
+
     @property
     def extra_state_attributes(self):
         """Return the state attributes."""
-        if not self.info:
+        if not self.hass.data[DOMAIN][self.controller_id][CONF_INFO]:
             return {}
-        return {
+        attributes = {
             "ihc_id": self.ihc_id,
             "ihc_name": self.ihc_name,
             "ihc_note": self.ihc_note,
             "ihc_position": self.ihc_position,
         }
+        multicontroller: bool = len(self.hass.data[DOMAIN]) > 1
+        # We only want to show the controller id if we have more than one
+        if multicontroller:
+            attributes["ihc_controller"] = self.controller_id
+        return attributes
 
     def on_ihc_change(self, ihc_id, value):
         """Handle IHC resource change.
 
         Derived classes must overwrite this to do device specific stuff.
         """
         raise NotImplementedError
+
+    @property
+    def device_info(self):
```

Comment:
We shouldn't add this until there's config entry support since it's a noop until then.

Reply:
But it is not possible to have config entry support without the config flow I guess?
I can remove the device_info function for now and add it with the config flow PR

Reply:
No, config entries require a config flow. 👍

Reply:
Should I remove the device_info function from this PR?

Reply:
I suggested this as new information was being added for use in device info and so i thought it made sense to actually set device info and use it rather than leaving it dangling.  Apologies, since it sounds like this was the wrong direction.  @MartinHjelmare i was trying to figure out ways to suggest https://github.com/home-assistant/core/pull/50957 be split up into smaller PRs to facilitate a faster review, so if you spot it would be helpful since i may have suggested the wrong approach.

Reply:
I'd start with minimums required for config flow and leave device info related things for later.

Reply:
I have removed the device_info function